### PR TITLE
Avoid two calls to put in enum overload of formatValueImpl

### PR DESCRIPTION
**Pull request by nordlow**
Follow-up to https://github.com/dlang/phobos/pull/8593.

### CODE REVIEW

File: std/format/internal/write.d [hunk 1]
```diff
@@ -3031,9 +3031,8 @@ if (is(T == enum))
         auto w2 = appender!string();
 
         // val is not a member of T, output cast(T) rawValue instead.
-        put(w2, "cast(");
-        put(w2, T.stringof);
-        put(w2, ")");
+        static immutable prefix = "cast(" ~ T.stringof ~ ")";
```
Comment (nordlow) [inline std/format/internal/write.d:3034]: Ok with me to use enum here instead or inline the reference to `prefix`.

Comment (thewilsonator) [reply std/format/internal/write.d:3034]: enum is good.

**Review by thewilsonator — COMMENTED** [review]
(Not that out need to do it in this PR, but) This whole function should be rewritten as 
```d
    import std.array : appender;
    import std.range.primitives : put;
    if (f.spec != 's')
       formatValueImpl(w, cast(OriginalType!T) val, f);
    ...
```

it currently has the early return the wrong way around.

File: std/format/internal/write.d (hunk 1 above)
Comment (nordlow) [reply std/format/internal/write.d]: Fixed.

**Comment by nordlow** [comment]
> (Not that out need to do it in this PR, but) This whole function should be rewritten as
> 
> ```d
>     import std.array : appender;
>     import std.range.primitives : put;
>     if (f.spec != 's')
>        formatValueImpl(w, cast(OriginalType!T) val, f);
>     ...
> ```
> 
> it currently has the early return the wrong way around.

Resolved via 489dad4fb2754b8dffefebe749cb946271810dab.

**Comment by atilaneves** [comment]
I don't understand what this PR is trying to do.

**Comment by nordlow** [comment]
> I don't understand what this PR is trying to do.

Oops, I managed to pull in changes from another PR. Will adjust.

**Comment by nordlow** [comment]
> it currently has the early return the wrong way around.

See https://github.com/dlang/phobos/pull/8600.

**Comment by atilaneves** [comment]
> > I don't understand what this PR is trying to do.
> 
> It replaces three calls to `put` with a single call to `put` plus a compile-time string concat.

And this is good because...?

**Comment by nordlow** [comment]
> > It replaces three calls to `put` with a single call to `put` plus a compile-time string concat.
> 
> And this is good because...?

I thought it's obvious. Faster run-time execution.

**Pull request merged**